### PR TITLE
CLI: header and verbose support

### PR DESCRIPTION
**Pull request by DrMegavolt**

### CODE REVIEW

File: bin/eg-generator.js [hunk 1]
```diff
@@ -8,7 +8,14 @@ module.exports = class EgGenerator extends Generator {
     this._configuration = null;
     this.eg = this.env.eg;
     this.argv = this.env.argv;
-    this.admin = require('../lib/admin')(config.gatewayConfig.admin);
+    if (this.argv) {
+      let cliConfig = config.systemConfig.cli || {};
```
Comment (kevinswiber) [inline bin/eg-generator.js]: Could we look for something like an `EG_ADMIN_URL` environment variable and fall-back to `config.gatewayConfig.admin` if it doesn't exist?

Comment (DrMegavolt) [inline bin/eg-generator.js]: I'll add EG_ADMIN_URL, but we should not add `config.gatewayConfig.admin`
'cuz config may be different, for example if admin API is protected by EG.
Also `config.cli` has additional property `protocol`, that doesn't make sense for REST admin

Comment (kevinswiber) [inline bin/eg-generator.js]: Okie dokie.  :+1:

File: lib/admin/index.js [hunk 2]
```diff
@@ -1,13 +1,15 @@
 // TODO: this is EG admin client; internal utility
-module.exports = function (adminConfig) {
-  let host = adminConfig.hostname || 'localhost';
-  let port = adminConfig.port || 9876;
-  adminConfig.baseUrl = adminConfig.baseUrl || `http://${host}:${port}/`;
+module.exports = function ({cliConfig, headers, verbose}) {
+  let host = cliConfig.hostname || 'localhost';
+  let port = cliConfig.port || 9876;
+  let protocol = cliConfig.protocol || 'http';
+  let baseUrl = cliConfig.baseUrl || `${protocol}://${host}:${port}/`;
```
Comment (kevinswiber) [inline lib/admin/index.js]: Could we just do a single `url` parameter in the `cliConfig` instead of `protocol`, `host`, and `port`?

**Comment by DrMegavolt** [comment]
@kevinswiber
 Lookup priority
1) EG_ADMIN_URL 
2) `cli.url`
3) http://localhost:9876 default

**Pull request merged**